lms/static/scripts/ui-playground/components/ErrorComponents.js

```diff
@@ -42,29 +44,390 @@ function ErrorDialogExample() {
   }
 }
 
+function ErrorDialogAppExample({ errorCode = 'reused_consumer_key' }) {
+  const config = { api: {}, errorDialog: { errorCode } };
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  if (!dialogOpen) {
+    return (
+      <LabeledButton
+        onClick={() => setDialogOpen(!dialogOpen)}
+        variant="primary"
+      >
+        Show ErrorDialogApp Example
+      </LabeledButton>
+    );
+  } else {
+    return (
+      <Config.Provider value={config}>
+        <ErrorDialogApp />
+      </Config.Provider>
+    );
+  }
+}
+
+function OAuth2RedirectErrorAppExample({ errorCode = '' }) {
+  const config = {
+    api: {},
+    OAuth2RedirectError: {
+      errorCode,
+      canvasScopes: [
+        'This is a fake Canvas scope',
+        'These scopes would be',
+        'Written to the JS configuration object',
+        'By the back end in real life',
+      ],
+      errorDetails: fakeDetails,
+    },
+  };
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  if (!dialogOpen) {
+    return (
+      <LabeledButton
+        onClick={() => setDialogOpen(!dialogOpen)}
+        variant="primary"
+      >
+        Show OAuth2RedirectErrorApp Example
+      </LabeledButton>
+    );
+  } else {
+    return (
+      <Config.Provider value={config}>
+        <OAuth2RedirectErrorApp />
+      </Config.Provider>
+    );
+  }
+}
+
 export default function ErrorComponents() {
   return (
     <Library.Page title="Errors">
+      <div className="LMSLibrary__content">
+        <h2>Intro</h2>
+        <p>To understand errors it is useful to understand:</p>
+        <h3>Modes</h3>
+        <ul>
+          <li>
+            The front-end application operates in one of <b>four {'"modes"'}</b>
```

Comment:
I think some of the specifics are liable to get out of date here, eg. the fact that there are currently four modes or the details of specific errors. The general elements seem likely to persist into the future though:

- The different triggers that can lead to errors being displayed
- The elements of a backend-provided error
- The different kinds of error that the frontend may need to display
- The different error components
- What the error components present